sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -226,7 +227,8 @@ case class RelationConversions(
             tableDesc.partitionColumnNames.isEmpty && isConvertible(tableDesc) &&
             SQLConf.get.getConf(HiveUtils.CONVERT_METASTORE_CTAS) =>
         // validation is required to be done here before relation conversion.
-        DDLUtils.checkDataColNames(tableDesc.copy(schema = query.schema))
+        val tableSchema = CharVarcharUtils.replaceCharVarcharWithStringInSchema(query.schema)
+        DDLUtils.checkDataColNames(tableDesc.copy(schema = tableSchema))
```

Comment:
do we really need to restore char/varchar for this `checkDataColNames`?

Reply:
Oh , my mistake